rebasebot/bot.py

```diff
@@ -162,30 +162,12 @@ def _do_rebase(gitwd, source, dest, source_repo, tag_policy):
                 raise RepoException(f"Git rebase failed: {ex}") from ex
 
 
-def _needs_merge(gitwd, dest):
-    logging.info("Checking if we need a merge commit")
-
-    try:
-        gitwd.git.checkout(f"dest/{dest.branch}")
-        gitwd.git.merge("--no-ff", "rebase")
-    except git.GitCommandError:
-        logging.info("Merge commit is required")
-        return True
-    finally:
-        gitwd.git.reset("--hard", "HEAD")
-        gitwd.git.checkout("rebase")
-
-    logging.info("Merge commit is not required")
-
-    return False
-
-
 def _do_merge(gitwd, dest):
     logging.info("Performing merge")
     try:
         gitwd.git.merge(
-            f"dest/{dest.branch}", "-Xours", "-m",
-            f"UPSTREAM: <carry>: Merge branch '{dest.branch}' in {gitwd.active_branch}"
+            f"dest/{dest.branch}", "-m",
+            f"merge branch '{dest.branch}' in {gitwd.active_branch}"
```

Comment:
Now that this happens first, is there a way we can tell this to always prefer changes from the upstream in the case of conflicts?

Is this definitely doing the exact same thing? As far as I could tell the rebase method basically resets the branch so the file tree looks exactly like the upstream, is there a chance we have files left over after this merge?